### ISSUE TITLE
[Request]: Cancel CI jobs when new commit pushed to PR

### ISSUE THREAD

**Issue by miquelbeltran** [fluttercommunity/plus_plugins#1145]
### Plugin

all (CI)

### Use case

This repo contains long-running jobs that are triggered each time a commit is pushed.

When submitting a PR, each time the code is pushed the jobs will start, but the previous won't be canceled, causing multiple jobs to "use" all available virtual machines and slowing development process.

### Proposal

Configure CI jobs, so they get cancelled when a new commit is pushed in the same PR.

Note that this shouldn't cancel jobs running on other branches/PRs, only on the current branch.

**Comment by miquelbeltran** [comment]
I think that adding 

```
concurrency:
  group: ${{ github.head_ref || github.ref_name }} 
  cancel-in-progress: true
```

on all yaml files would work.

`github.head_ref || github.ref_name` will use the branch name as unique group id, so we can have multiple PRs on different banches running

**Comment by vbuberen** [comment]
I missed this issue initially, but why not use the out of the box ready solution?
https://github.com/styfle/cancel-workflow-action

I use it on some other projects and it works fine.

**Comment by miquelbeltran** [comment]
could have been an option, yes, let's keep it in mind if the `concurrency` setting does not behave as expected.

I forgot to close this, as it has been solved in the #1147 PR
